model/common/src/icon4py/model/common/decomposition/mpi_decomposition.py

```diff
@@ -423,3 +425,19 @@ def create_multinode_node_exchange(
         return GHexMultiNodeExchange(props, decomp_info)
     else:
         return SingleNodeExchange()
+
+
+class GlobalReductions(Reductions):
+    def __init__(self, props: definitions.ProcessProperties):
+        self._props = props
+
+    def min(self, buffer: data_alloc.NDArray, array_ns: ModuleType = np) -> state_utils.ScalarType:
+        local_min = array_ns.min(buffer)
+        recv_buffer = array_ns.empty(1, dtype=buffer.dtype)
+        self._props.comm.Allreduce(local_min, recv_buffer, mpi4py.MPI.MIN)
```

Comment:
We should probably consider the stream synchronisation issue here for cupy arrays. See the example in the tutorial: https://mpi4py.readthedocs.io/en/stable/tutorial.html#gpu-aware-mpi-python-gpu-arrays 
I did not test it on GPU, did you? I would at least leave an TODO here that this might become a problem. Also ask the HPC - MPI pros like @msimberg ...